### ISSUE TITLE
Sync AWX Project with Git fails with a different "SCM BRANCH/TAG/COMMIT" than the default repository branch

### ISSUE THREAD

**Issue by khelkun** [ansible/awx#2088]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
I use AWX Project synced with a Git repository with SSH source control credential type which uses a private key and a passphrase.
Everything's fine except if I try to sync the AWX Project with a different branch than the default branch of the Git repository e.g: using the "SCM BRANCH/TAG/COMMIT" field in the AWX Project configuration. In this use case the project sync job launches the command "/usr/bin/git ls-remote origin -h refs/heads/mybranch" after clonging the repository branch. And this command apparently fails to authenticate through SSH with the private key and passphrase.

##### ENVIRONMENT
* AWX version: 1.0.6.3
* AWX install method: docker on linux
* Ansible version:  2.5.2
* Operating System: Ubuntu 16.04 LTS
* Web Browser: Google Chrome 67.0.3396.99 (Official Build) (64-bit)

##### STEPS TO REPRODUCE
- We want to sync an AWX project with a specific branch of Git (Gitlab in my case) repository. The authentication method to git is SSH with private key and passphrase.
- Create a credential of type "Source Control" with:
  - a username ("git") in my case
  - no password
  - a private key and its passphrase
- Create an AWX Project with:
  - "SCM Type" set to "Git" and sync it with a repository
  - "SCM Credential" set to the previously created "Source Control" credential
  - "SCM BRANCH/TAG/COMMIT" set to a branch different from the default clone branch of the Git repository, lets call it "mybranch"
- Save the project click on the button "Get latest SCM revision" to get the project from the Git repository

##### EXPECTED RESULTS
The project is successfully synced and exists in /var/lib/awx/project directory inside the awx_task_1 Docker container.

##### ACTUAL RESULTS
The Project sync job fails with the following error:
```
fatal: [localhost]: FAILED! => 
{
   "changed":false,
   "cmd":"/usr/bin/git ls-remote origin -h refs/heads/mybranch",
   "msg":"ssh: connect to host gitlab.mydomain.com port 22: Connection refused\\r\\nfatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.",
   "rc":128,
   "stderr":"ssh: connect to host gitlab.mydomain.com port 22: Connection refused\\r\\nfatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.\\n",
   "stderr_lines":[
      "ssh: connect to host gitlab.mydomain.com port 22: Connection refused",
      "fatal: Could not read from remote repository.",
      "",
      "Please make sure you have the correct access rights",
      "and the repository exists."
   ],
   "stdout":"",
   "stdout_lines":[

   ]
}
```
However:
- The project actually exists inside `/var/lib/awx/project/_X__myproject`
- a `git status` inside `/var/lib/awx/project/_X__myproject` reports that:
```
On branch mybranch
Your branch is up to date with 'origin/mybranch'.

nothing to commit, working tree clean
```

So as stated in my brief summary, the clone was done successfully but the command "/usr/bin/git ls-remote origin -h refs/heads/mybranch" apparently fails to authenticate through SSH with the private key and passphrase.

##### ADDITIONAL INFORMATION
This is a minor issue ihmo because if I leave the "SCM BRANCH/TAG/COMMIT" field empty in the Project configuration then everything goes fine. So as a workaround, I simply changed the default branch of my repository from "master" to "mybranch".
However @matburt on IRC told me I could file the issue here.

**Comment by YunfanZhang42** [comment]
Hi @khelkun thanks for your feedback on Ansible AWX project. Could you please provide some details on your AWX installation to help us investigate this issue? In particular, 

- Are you using a clustered installation for AWX?
- Does your SSH Git server support anonymous access (i.e. you can clone public repositories without setting up a SSH key)

**Comment by khelkun** [comment]
@YunfanZhang42 
> Are you using a clustered installation for AWX?

No

> Does your SSH Git server support anonymous access (i.e. you can clone public repositories without setting up a SSH key)

No, it's a private gitlab-ce (community edition) installation

The whole output of the project sync job is:
```
Enter passphrase for /tmp/awx_496_IIPjkX/credential_5: 
Identity added: /tmp/awx_496_IIPjkX/credential_5 (/tmp/awx_496_IIPjkX/credential_5)
Using /etc/ansible/ansible.cfg as config file

PLAY [all] *********************************************************************


TASK [delete project directory before update] **********************************
changed: [localhost]

TASK [check repo using git] ****************************************************
skipping: [localhost]

TASK [update project using git] ************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "/usr/bin/git ls-remote origin -h refs/heads/mybranch", "msg": "ssh: connect to host gitlab.mydomain.com port 22: Connection refused\\r\\nfatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.", "rc": 128, "stderr": "ssh: connect to host gitlab.mydomain.com port 22: Connection refused\\r\\nfatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.\\n", "stderr_lines": ["ssh: connect to host gitlab.mydomain.com port 22: Connection refused", "fatal: Could not read from remote repository.", "", "Please make sure you have the correct access rights", "and the repository exists."], "stdout": "", "stdout_lines": []}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=1   
```

**Comment by YunfanZhang42** [comment]
Hi @khelkun, we just released awx 1.0.7 today and we were unable to reproduce this bug on this version. Can you try to use awx 1.0.7 and see if this problem persists? Thanks!

**Comment by khelkun** [comment]
Hi @YunfanZhang42 I'll try. I don't see database changes in hte 1.0.7 so I assume I can upgrade from my 1.0.6.2 without any issue with my existing configuration with:
```bash
docker-compose pull && docker-compose up --force-recreate
```

**Comment by khelkun** [comment]
Hello @YunfanZhang42, sorry for the delay. I've just tested the 1.0.7. I still can reproduce the issue. 
The project repository is cloned in `/var/lib/awx/projects/_X__myproject` but the `git ls-remote origin -h refs/heads/mybranch` still fails.
It may not be relevant but I tried this:
```
docker exec -it awx_task_1 bash
[root@awx awx]# cd projects/_X__myproject
[root@awx _X__myproject]# git ls-remote origin -h refs/heads/mybranch
git@gitlab.evolucare.com's password:
```
So the git command prompts me for password. Ihmo:
- I use an AWX SCM credential which is a SSH private key (with passphrase) to authenticate on my Git origin
- Whether "SCM BRANCH/TAG/COMMIT" is setor not, the `git clone` of the project sync job succeeds to use this AWX SCM credential
- When "SCM BRANCH/TAG/COMMIT" is not set, I think AWX doesn't attempt to run the `git ls-remote origin` command, or may be it does but it succeeds to use the AWX SCM credential to run it.
- But when "SCM BRANCH/TAG/COMMIT" is set, AWX attempts to run `git ls-remote origin -h refs/heads/mybranch` without using the AWX SCM credential, or it fails to use it.
